test/functional/test_training_framework.py

```diff
@@ -75,17 +88,17 @@ def framework_training_fn():
             model.save(model_file)
 
 
-@pytest.mark.parametrize('user_script', [USER_SCRIPT, USER_SCRIPT_WITH_SAVE])
-def test_training_framework(user_script):
+def test_training_framework_with_save():
```

Comment:
Why do we give up using pytest.mark.parametrize? Since there is likely to have more tests using more scripts, for example, trainer tests, I definitely prefer using pytest.mark.parametrize for more scalable style.

Does using pytest.mark.parametrize break caching?

Reply:
Talked offline. Will change this later.